src/evaluate/visualization.py

```diff
@@ -0,0 +1,224 @@
+        self.ax.xaxis.grid(**self.format_cfg["angle_ln_args"])
+
+        # ax1 is the duplicate of axes[0] (self.ax)
+        # Remove everything from ax1 except the plot itself
+        self.ax1.axis("off")
+        self.ax1.set_zorder(9)
+
+        # Create the outer labels for each variable
+        l, text = self.ax.set_thetagrids(angles, labels=variables)
+
+        # Beautify them
+        labels = [t.get_text() for t in self.ax.get_xticklabels()]
+        labels = [
+            "\n".join(
+                textwrap.wrap(
+                    label,
+                    self.format_cfg["theta_tick_lbls_txt_wrap"],
+                    break_long_words=self.format_cfg["theta_tick_lbls_brk_lng_wrds"],
+                )
+            )
+            for label in labels
+        ]
+        self.ax.set_xticklabels(labels, **self.format_cfg["theta_tick_lbls"])
+
+        for t, a in zip(self.ax.get_xticklabels(), angles):
+            if a == 0:
+                t.set_ha("center")
+            elif a > 0 and a < 180:
+                t.set_ha("left")
+            elif a == 180:
+                t.set_ha("center")
+            else:
+                t.set_ha("right")
+
+        self.ax.tick_params(axis="both", pad=self.format_cfg["theta_tick_lbls_pad"])
+
+    def _scale_data(self, data, ranges):
+        """Scales data[1:] to ranges[0]"""
+        for d, (y1, y2) in zip(data[1:], ranges[1:]):
+            assert (y1 <= d <= y2) or (y2 <= d <= y1)
+        x1, x2 = ranges[0]
+        d = data[0]
+        sdata = [d]
+        for d, (y1, y2) in zip(data[1:], ranges[1:]):
+            sdata.append((d - y1) / (y2 - y1) * (x2 - x1) + x1)
+        return sdata
+
+    def plot(self, data, *args, **kwargs):
+        """Plots a line"""
+        sdata = self._scale_data(data, self.ranges)
+        self.ax1.plot(self.angle, np.r_[sdata, sdata[0]], *args, **kwargs)
+        self.plot_counter = self.plot_counter + 1
+
+    def fill(self, data, *args, **kwargs):
+        """Plots an area"""
+        sdata = self._scale_data(data, self.ranges)
+        self.ax1.fill(self.angle, np.r_[sdata, sdata[0]], *args, **kwargs)
+
+    def use_legend(self, *args, **kwargs):
+        """Shows a legend"""
+        self.ax1.legend(*args, **kwargs)
+
+    def set_title(self, title, pad=25, **kwargs):
+        """Set a title"""
+        self.ax.set_title(title, pad=pad, **kwargs)
+
+
+def radar_plot(data, model_names, invert_range=[]):
+    """
+    `data`: list of `dict`s of metric + value pairs.
+        E.g. data = [{"accuracy": 0.9, "precision":0.8},{"accuracy": 0.7, "precision":0.6}]
+    `names`: list of `str`s with model names
+        E.g. names = ["model1", "model 2", ...]
+    `invert_range`: list of `str`s with the metrics to invert (in cases when lower is better, e.g. speed)
+        E.g. invert_range=["latency_in_seconds"]
+    """
+    if isinstance(data, list) is False:
+        raise ValueError("The input must be a list of dicts of metric + value pairs")
```

Comment:
I think we discussed removing that test and leave it to `pd.DataFrame` to determine if it is in the right format.